### PR TITLE
Write absolute AppDataPath to app.ini when installing

### PR DESCRIPTION
**Pull request by wxiaoguang**
If the APP_DATA_PATH isn't written into the config when installing, then its value is uncertain because some Gitea command doesn't run with correct WorkPath.

This is a quick fix for #25330  and can be backported.

### CODE REVIEW

**Review by lunny — APPROVED** [review]

File: modules/setting/server.go [hunk 1]
```diff
@@ -322,10 +322,15 @@ func loadServerFrom(rootCfg ConfigProvider) {
...
-	AppDataPath = sec.Key("APP_DATA_PATH").MustString(path.Join(AppWorkPath, "data"))
+	AppDataPath = sec.Key("APP_DATA_PATH").String()
+	if AppDataPath == "" {
+		AppDataPath = "data"
+	}
 	if !filepath.IsAbs(AppDataPath) {
-		log.Info("The provided APP_DATA_PATH: %s is not absolute - it will be made absolute against the work path: %s", AppDataPath, AppWorkPath)
 		AppDataPath = filepath.ToSlash(filepath.Join(AppWorkPath, AppDataPath))
+		if InstallLock {
+			log.Error("The provided APP_DATA_PATH is not absolute, you must set it to absolute like APP_DATA_PATH=%s", AppDataPath)
```
Comment (delvh) [inline modules/setting/server.go]: This error message doesn't make sense in its new form.
`AppDataPath` is the component that didn't work, right?
Why should we set it like the invalid path?`

Comment (delvh) [reply modules/setting/server.go]: Okay, just noticed the line above.

**Review by delvh — APPROVED** [review]

**Comment by wxiaoguang** [comment]
Thank you for reviewing and approving, I just realized that there are more details and this PR is not ideal for 1.21.

I would propose a new separate PR for 1.20 later. 🙏

**Comment by wxiaoguang** [comment]
OK, hard decision ..... let's just do this one-line patch, and leave more problems to #25330

**Review by silverwind — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Pull request merged**